### PR TITLE
Product gallery on detail page 

### PR DESCRIPTION
**Pull request by PiotrZaranek**

### CODE REVIEW

File: fe/projects/AspNetCore/package.json
Comment (SebaKas21) [inline fe/projects/AspNetCore/package.json]: **suggestion:** Please remove ^

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js:19]: **suggestion (non-blocking):** Redundant empty lines

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Is never used. if you want use, please change fabricsSliderOnEnd to meanfuling name for product detail (not fabrics)

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Is never used. if you want use, please change fabricsSliderOnStart to meanfuling name for product detail (not fabrics)

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion (non-blocking):** redundant empty line

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion (non-blocking):** Redundant space after semicolon

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Too much commented code. If is not needed, please remove.

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion:** Ccommented code. If is not needed, please remove.

File: fe/projects/AspNetCore/src/stories/20-Product.stories.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/stories/20-Product.stories.js]: **suggestion:** Change property to lower case

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion (non-blocking):** Please change to rem

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion (non-blocking):** Please change to rem

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion (non-blocking):** Please use color variable

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion (non-blocking):** You can move this to line: 154 for desktop-product-gallery

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Redundant empty line

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Recover other functions <-It is the most important

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js:12]: **suggestion:** If something is not used, remove import

**Review by SebaKas21 — CHANGES_REQUESTED** [review]

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Missing lazy-loading

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Missing lazy-loading

**Review by SebaKas21 — CHANGES_REQUESTED** [review]

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (dawiddworak88) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Please add a check whether images exist because if there are no images added the the product yet, the whole page will crash:

{props.images && props.images.length &&

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (dawiddworak88) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion (non-blocking):** please move these values to constants and give them descriptive names, let's avoid magic strings. Please, ask @SebaKas21 for hints about it

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (dawiddworak88) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** please remove all the code commented out completely. We use git and we can always find a history

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (dawiddworak88) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **question:** Where is the number 47% coming from?

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** give full function names, example: showMoreImages and setShowMoreImages

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** give full function names

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Use Button component from MUI library and make more readable code

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Move this className to lazyLoad and remove redundant div

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Move this className to lazyLoad and remove redundant div

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** check if photos are added. Take a look on line 147 or 172

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** Use Button component from MUI library and make more readable code

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion:** Use 0 instead of 0px;

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion:** Use rem instead of px;

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion (non-blocking):** I think we can remove it

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion (non-blocking):** I think we can minimize the class names

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion (non-blocking):** I think we can move this to &__arrow

**Review by SebaKas21 — CHANGES_REQUESTED** [review]

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** paramete imageSrc is wrong passed. Should be image.imageSrc instead of image.src or change imageViewModel on BE

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** paramete imageSrc is wrong passed. Should be image.imageSrc instead of image.src or change imageViewModel on BE

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion (non-blocking):** Please try without !important

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion (non-blocking):** Please try without !important

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss]: **suggestion (non-blocking):** Please try without !important

**Review by SebaKas21 — CHANGES_REQUESTED** [review]

**Comment by SebaKas21** [comment]
Hi @PiotrZaranek, you are on the right track 👍

File: be/src/Project/Web/Buyer/Buyer.Web/Areas/Products/ModelBuilders/Products/ProductDetailModelBuilder.cs
Comment (dawiddworak88) [inline be/src/Project/Web/Buyer/Buyer.Web/Areas/Products/ModelBuilders/Products/ProductDetailModelBuilder.cs]: **suggestion:** () is redundant, please replace with:
var imageViewModel = new ImageViewModel
{
...
}

File: be/src/Project/Web/Buyer/Buyer.Web/Areas/Products/ModelBuilders/Products/ProductDetailModelBuilder.cs
Comment (dawiddworak88) [inline be/src/Project/Web/Buyer/Buyer.Web/Areas/Products/ModelBuilders/Products/ProductDetailModelBuilder.cs]: **suggestion:** remove redundant ()

File: fe/projects/AspNetCore/src/areas/Products/components/ProductImageModal/ImageModal.js
Comment (dawiddworak88) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductImageModal/ImageModal.js]: **suggestion:** please remove all console.log

File: fe/projects/AspNetCore/src/areas/Products/components/ProductImageModal/ImageModal.js
Comment (dawiddworak88) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductImageModal/ImageModal.js]: **suggestion:** is useRef critical here? Can't it be avoided? It should be possible

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (dawiddworak88) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js:6]: **question and suggestion:** is the "react-image-gallery" used anywhere after removing it here? If not, let's remove it completely from dependencies (packages.json)

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js
Comment (dawiddworak88) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.js]: **suggestion:** please move '6' to constants and give it meaningful name. We should avoid 'magic strings' in code

File: fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductDetail/ProductDetail.scss:199]: **suggestion (non-blocking):** Let's try add to .product-detail

File: fe/projects/AspNetCore/src/areas/Products/components/ProductImageModal/ImageModal.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductImageModal/ImageModal.js:58]: **suggestion:** Missing propTypes. Let's look on ProductDetail line 327

File: fe/projects/AspNetCore/src/areas/Products/components/ProductImageModal/ImageModal.js
Comment (SebaKas21) [inline fe/projects/AspNetCore/src/areas/Products/components/ProductImageModal/ImageModal.js]: **suggestion (non-blocking):** Too much tabs

**Review by SebaKas21 — CHANGES_REQUESTED** [review]

**Review by SebaKas21 — APPROVED** [review]
Hi @PiotrZaranek,

In my opinion your work is well done, but we should wait for on the final decision @dawiddworak88

File: be/src/Project/Web/Buyer/Buyer.Web/Shared/ViewModels/Images/ImageViewModel.cs
Comment (SebaKas21) [inline be/src/Project/Web/Buyer/Buyer.Web/Shared/ViewModels/Images/ImageViewModel.cs:9]: **suggestion:** Missing ImageAlt in BE

**Review by SebaKas21 — APPROVED** [review]
Nice 👍

**Review by SebaKas21 — CHANGES_REQUESTED** [review]
Hi @PiotrZaranek

There is no check if there are more than one image, 
if there is only one, it should be displayed in full width

[EDIT] Other than that, I found no other bugs

obraz

**Review by dawiddworak88 — APPROVED** [review]

**Pull request merged**
